src/main/java/mil/nga/tiff/FileDirectory.java

```diff
@@ -1274,23 +1274,23 @@ private Number readValue(ByteReader reader, FieldType fieldType) {
 			value = reader.readDouble();
 			break;
 		default:
-			throw new TiffException("Unsupported raster field type: "
-					+ fieldType);
+			throw new TiffException("Unsupported raster sampleType type: "
+					+ sampleType);
 		}
 
 		return value;
 	}
 
 	/**
-	 * Get the field type for the sample
-	 * 
+	 * Get the raster sample type for the sample @see Rasters.SampleType
+	 *
 	 * @param sampleIndex
 	 *            sample index
-	 * @return field type
+	 * @return Sample type
 	 */
-	public FieldType getFieldTypeForSample(int sampleIndex) {
+	public Rasters.SampleType getFieldTypeForSample(int sampleIndex) {
```

Comment:
Please rename to getTypeForSample or getSampleTypeForSample (field and sample types are now different thing)